src/Shared/FileUtilities.cs

```diff
@@ -966,6 +966,19 @@ internal static bool FileExistsNoThrow(string fullPath, IFileSystem fileSystem =
             }
         }
 
+        /// <summary>
+        /// Get all files from directory matching pattern
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="pattern"></param>
+        /// <param name="fileSystem"></param>
+        /// <returns>list of files or null if it is impossible to get files</returns>
+        internal static string[] DirectoryGetFiles(string path, string pattern = "*", IFileSystem fileSystem = null)
+        {
+            path = AttemptToShortenPath(path);
```

Comment:
can you confirm ```AttemptToShortenPath``` handles NULL or path.Length = 0 cases?  If not, should this method validate ```path```?

Reply:
I used this method at some places and for same reason as it was used in `FileExistsNoThrow`, but no, this function do not test nullOrEmpty. I have not looked it, but I believe this is handled somewhere upper in call stack.

Reply:
It might be reasonable to check and make sure there are tests for that, as that sort of behavior could be changed in the future, possibly by accident.

Reply:
null is handled here:
https://github.com/dotnet/msbuild/blob/5190c1336185d67cb59a69ea9c47cf89e64698b4/src/Tasks/AssemblyDependency/Resolver.cs#L347
However I will write some unit tests to cover these use cases with current behavior.